### PR TITLE
Dec 397 date metadata validation

### PR DESCRIPTION
**Pull request by rfox2**
**What**: Added validation for date metadata fields
**How**: I merged in Jon's changes from another branch he had been working on which added in the JSON metadata fields, added the class structure for validation; I didn't work on the UI at all to test validation, nor did I write any feature tests;

### CODE REVIEW

File: app/assets/javascripts/components/forms/DateField.js.jsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          re.lastIndex++;
+      }
+      return m;
+    }
+
+  },
+
+  render: function () {
+    return (
+        <FormRow id={this.formId()} type="string" required={this.props.required} title={this.props.title} help={this.props.help} errorMsg={this.props.errorMsg} >
```
Comment (jaronkk) [inline app/assets/javascripts/components/forms/DateField.js.jsx]: It looks like this whole section is indented with an extra space

File: app/assets/javascripts/components/forms/DateField.js.jsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
+//app/assets/javascripts/components/forms/TextField.jsx
```
Comment (danthewolfe) [inline app/assets/javascripts/components/forms/DateField.js.jsx]: You're going to need to change this to add require("react") and module.exports="DateField"

Also rename to DateField.jsx to be consistent with other file names.

**Comment by jaronkk** [comment]
:+1:

**Comment by danthewolfe** [comment]
:+1:

**Pull request merged**